src/test/resources/project-builder-complex/complex-user/dao/dao.go

```diff
@@ -194,3 +250,64 @@ func (dao *DAO) IdentifyComplexUser(input IdentifyComplexUserInput) (*ComplexUse
 
 	return &complexUser, nil
 }
+
+// CreateTempleUser creates a new templeUser in the datastore, returning the newly created templeUser
+func (dao *DAO) CreateTempleUser(input CreateTempleUserInput) (*TempleUser, error) {
+	row := executeQueryWithRowResponse(dao.DB, "INSERT INTO temple_user (id, int_field, double_field, string_field, bool_field, date_field, time_field, date_time_field, blob_field) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9) RETURNING id, int_field, double_field, string_field, bool_field, date_field, time_field, date_time_field, blob_field;", input.ID, input.ParentID, input.IntField, input.DoubleField, input.StringField, input.BoolField, input.DateField, input.TimeField, input.DateTimeField, input.BlobField)
```

Comment:
Ffs, I’ve been running this from templefiles, but this code is generated from a unit test: the input is post-validate stage, and wasn’t valid. Any idea how to prevent cases like that? Just manually checking every unit test every time the validator is updated?

Reply:
Hmm not really too sure. I guess this is an issue with where we inject the `ParentID` attribute. It feels as though it shouldn't be added in the validator stage, but rather it's implicit from the actual thing you're trying to generate. That way it's not possible to construct an invalid object to generate using the available constructors.